hs_core/views/utils.py

```diff
@@ -786,34 +786,23 @@ def rename_irods_file_or_folder_in_django(resource, src_name, tgt_name):
     src_folder, base = ResourceFile.resource_path_is_acceptable(resource, src_name,
                                                                 test_exists=False)
     tgt_folder, _ = ResourceFile.resource_path_is_acceptable(resource, tgt_name, test_exists=False)
-    file_move = src_folder != tgt_folder
+    file_or_folder_move = src_folder != tgt_folder
     try:
         res_file_obj = ResourceFile.get(resource=resource, file=base, folder=src_folder)
-        # if the source file is part of a FileSet, we need to remove it from that FileSet in the
-        # case file being moved
-        if file_move and resource.resource_type == 'CompositeResource':
-            if res_file_obj.has_logical_file and res_file_obj.logical_file.is_fileset:
-                try:
-                    aggregation = resource.get_aggregation_by_name(res_file_obj.file_folder)
-                    if aggregation.is_fileset:
-                        # remove aggregation form the file
-                        res_file_obj.logical_file_content_object = None
-                        res_file_obj.save()
-                except ObjectDoesNotExist:
-                    pass
+        # if the source file is part of a FileSet or Model Program/Instance aggregation (based on folder),
+        # we need to remove it from that aggregation in the case the file is being moved out of that aggregation
+        if file_or_folder_move and resource.resource_type == 'CompositeResource':
+            resource.remove_aggregation_from_file(res_file_obj, src_folder, tgt_folder)
```

Comment:
I like how this was consolidated but I'd like to take it even further.  I think the aggregation should hook into the the set_storage_path and do what's needed there.